api.php

```diff
@@ -1,12 +1,22 @@
 <?php
 require_once('Books.php');
-if (preg_match("[^/api/books(|/)$]", $_SERVER['REQUEST_URI'])) {
-    Books::getInstance()->allBooks();
+if (!isset($_SERVER['PHP_AUTH_USER'])) {

```

Comment:
What if execute here login and in case of incorrect login redirect user, so it won't be needed to use quotes in next code.

like:
function get(&$value, $default = null)
{
    return isset($value) ? $value : $default;
}

$user = get($_SERVER['PHP_AUTH_USER'], false)
$pass = get($_SERVER['PHP_AUTH_PW'], false)
if (!$user or !$pass) {
    redirect
}
$login = Books::getInstance()->auth($user, $pass)
if (!$login) {
    redirect
}